### DISCUSSION TITLE
How to get the user's server/guild specific "display_name"?

### DISCUSSION THREAD

**Discussion by deltamelter** [Rapptz/discord.py#9625] [Q&A] [answered]
my bot can see the users and their name with discriminator, but this is usually totally different to what is labelled in discord as Display Name
How do I get this display name which can be different on each server?

**Comment by LeoCx1000** [comment]
There is the [`Member.display_name`](https://discordpy.readthedocs.io/en/stable/api.html#discord.Member.display_name) attribute for that.

**Reply by deltamelter** [reply]
this just displays the same username without discriminator, ie the username that appears underneath the name I want. 
Do I have to use specific intents or something?

**Reply by No767** [reply]
Perhaps are you looking for `Member.global_name`?

**Comment by deltamelter** [comment]
> Perhaps are you looking for `Member.global_name`?

global_name is the same, it is the actual username, not the server specific display_name

**Comment by Soheab** [comment]
Do you perhaps want [`Member.nick`](https://discordpy.readthedocs.io/en/stable/api.html#discord.Member.nick)

- `name`: unique username
- `global_name`: "display name" in the UI
- `nick`: guild specific name
- `display_name`: nick or global_name or name (like the UI)
- `discriminator`: discrim or "0" if migrated

**Reply by deltamelter** [reply]
Member.nick is "None"

myServerUsername
myusername#0
`
<Member id=36416???????7546 name='myusername' discriminator='0' bot=False nick=None guild=<Guild id=379????????711 name='time' shard_id=0 chunked=True member_count=2>>
`
`bot.guilds[1].members[1].display_name` = `myusername`
`bot.guilds[1].members[1].name` = `myusername`

**Reply by Soheab** [reply]
> Member.nick is "None"
> 
> myServerUsername
> myusername#0
> `
> <Member id=36416???????7546 name='myusername' discriminator='0' bot=False nick=None guild=<Guild id=379????????711 name='time' shard_id=0 chunked=True member_count=2>>
> `
> `bot.guilds[1].members[1].display_name` = `myusername`
> `bot.guilds[1].members[1].name` = `myusername`
> 

1. You shouldn't really rely on a `__repr__`.
2. Looks like that specific member doesn't have a nickname in that specific server.

Consider using the [discord.py server](https://discord.gg/dpy) for further help.

**Reply by deltamelter** [reply]
> You shouldn't really rely on a __repr__.

I'm not really, it still returns None
> Looks like that specific member doesn't have a nickname in that specific server.

I haven't set a specific server nick name no, but it using the a different display_name than the username, a default one that was set at migration.
Just changed the server specific display name but I can't see this still with the bot
![image](https://github.com/Rapptz/discord.py/assets/27061628/bbd5de5f-1c70-4133-9a90-eb4194475e20)


> Consider using the [discord.py server](https://discord.gg/dpy) for further help.

I'll have a look

**Reply by Soheab** [reply]
> > You shouldn't really rely on a __repr__.
> 
> I'm not really, it still returns None
> > Looks like that specific member doesn't have a nickname in that specific server.
> 
> I haven't set a specific server nick name no, but it using the a different display_name than the username, a default one that was set at migration.
> Just changed the server specific display name but I can't see this still with the bot
> ![image](https://github.com/Rapptz/discord.py/assets/27061628/bbd5de5f-1c70-4133-9a90-eb4194475e20)
> 
> 
> > Consider using the [discord.py server](https://discord.gg/dpy) for further help.
> 
> I'll have a look

There is no such thing as a "server display name", that's your `.global _name` which everyone on discord can see.

There is something called a "nickname" which is server specific. Make sure you're in the correct tab, "server profiles".

**Reply by deltamelter** [reply]
Ok, maybe it is not called the "server display name", but I can't access this name that "everyone on discord can see", my bot cannot see this.
That is what I am trying to get to.  If this is the `global_name`,then how do I get this from the context of `ctx.message.author`?

**Reply by Soheab** [reply]
> Ok, maybe it is not called the "server display name", but I can't access this name that "everyone on discord can see", my bot cannot see this.
> That is what I am trying to get to.  If this is the `global_name`,then how do I get this from the context of `ctx.message.author`?

Should be `ctx.author.global_name`

**Reply by deltamelter** [reply, marked as answer]
it's version specific it seems

> Installing collected packages: discord.py, discord
  Attempting uninstall: discord.py
    Found existing installation: discord.py 2.2.2
    Uninstalling discord.py-2.2.2:
      Successfully uninstalled discord.py-2.2.2
Successfully installed discord-2.3.2 discord.py-2.3.2

global_name and nick populated now.

**Reply by ignxdank** [reply]
The new name changes were supported by the library starting in v2.3, so that would be why
